libControls/ToolTip.cpp

```diff
@@ -102,6 +116,14 @@ void ToolTip::draw() const
 		auto& renderer = NAS2D::Utility<NAS2D::Renderer>::get();
 		renderer.drawBoxFilled(rect(), NAS2D::Color::DarkGray);
 		renderer.drawBox(rect(), NAS2D::Color::Black);
-		renderer.drawText(mFont, mFocusedControl->second, position() + PaddingSize);
+
+		std::istringstream stream(mFocusedControl->second);
+		std::string line;
+		auto linePosition = position() + PaddingSize;
+		while (std::getline(stream, line, '\n'))
+		{
+			renderer.drawText(mFont, line, linePosition);
+			linePosition.y += mFont.height();
+		}

```

Comment:
This seems pretty close to duplicated code. Effectively there is a `forEachLine` functionality that has been duplicated, though with different details on what to do with those lines.